cli/src/main/java/org/jboss/gm/cli/Main.java

```diff
@@ -219,19 +218,16 @@ private void executeGradle() throws ManipulationException {
             build.setStandardOutput(System.out);
             build.setStandardError(System.err);
             build.run();
-        } catch (BuildException e) {
-            logger.error("Caught exception running build", e.getCause());
-            throw new ManipulationException("Caught exception running build", e.getCause());
-        } catch (GradleConnectionException e) {
+        } catch (RuntimeException e) {
             // Unable to do instanceof comparison due to different classloader
-            if (e.getCause().getClass().getName().equals("org.gradle.api.UncheckedIOException")) {
+            if (e.getCause() != null && "org.gradle.api.UncheckedIOException".equals(e.getCause().getClass().getName())) {
                 logger.debug("Hit https://github.com/gradle/gradle/issues/9339 ", e);
                 logger.error(
                         "Build exception but unable to transfer message due to mix of JDK versions. Examine log for problems");
             } else {
                 logger.error("Gradle connection exception", e);
             }
-            throw new ManipulationException("Problem executing build");
+            throw new ManipulationException("Problem executing build", e);
```

Comment:
Marking this section for discussion (so I don't forget)

Reply:
I'm reverting as much as I can here.

The issue is we have three different possible code paths/messages here depending on the Gradle version so it's not as simple as just checking the exception type or message---the code still works (in the sense that an exception is thrown here), but it's hard to test, especially since we're printing slightly different messages in each case.

Reply:
I've been investigating the different exceptions and the failing 4.10 test with restURL ; its subtle to do with prior tests (runAlignment) influencing latter due to cached objects when the plugin is run within the daemon ... I think - essentially the ConfigCache _within_ the AlignmentTask is carrying over which is interfering.